pkg/expression/util.go

```diff
@@ -1450,6 +1450,50 @@ func ContainCorrelatedColumn(exprs []Expression) bool {
 	return false
 }
 
+// ProjectionBenefitsFromPushedDown evaluates if the expressions can improve performance when pushed down to TiKV
+// Projections are not pushed down to tikv by default, thus we need to check strictly here to avoid potential performance degradation.
+// Note: virtual column is not considered here, since this function cares performance instead of functionality
+func ProjectionBenefitsFromPushedDown(exprs []Expression, inputSchemaLen int) bool {
+	allColRef := true
+	colRefCount := 0
+	for _, expr := range exprs {
+		switch v := expr.(type) {
+		case *Column:
+			colRefCount = colRefCount + 1
+			continue
+		case *ScalarFunction:
+			allColRef = false
+			switch v.FuncName.L {
+			case ast.JSONDepth, ast.JSONLength, ast.JSONType, ast.JSONValid, ast.JSONContains, ast.JSONContainsPath,
+				ast.JSONExtract, ast.JSONKeys, ast.JSONSearch, ast.JSONMemberOf, ast.JSONOverlaps:
+				continue
+			case ast.JSONUnquote:
+				arg0 := v.GetArgs()[0]
+				// Only `->>` which parsed to JSONUnquote(CAST(JSONExtract() AS string)) can be pushed down to tikv
+				if fChild, ok := arg0.(*ScalarFunction); ok {
```

Comment:
maybe we can use a function like `JSONUnquoteBenefitsFromPushDown` to make the code more clear.

Reply:
Nice suggestion, updated.